onnxscript/evaluator.py

```diff
@@ -115,59 +113,74 @@ class Evaluator(abc.ABC):
     supported by onnxscript to those expected by a particular backend.
     """
 
-    def eval(self, schema, inputs, attributes):
-        closure = self.adapt_attributes(schema, attributes)
+    def eval(
+        self, schema: onnx.defs.OpSchema, inputs: Sequence[Any], attributes: Mapping[str, Any]
+    ):
+        attributes, closure = self.adapt_attributes(schema, attributes)
         inputs = self.adapt_inputs(schema, inputs)
         outputs = self._eval(schema, inputs, attributes, closure)
         return self.adapt_outputs(schema, outputs)
 
-    def adapt_inputs(self, schema, inputs):
+    def adapt_inputs(self, schema: onnx.defs.OpSchema, inputs: Sequence[Any]):
         """Transform inputs to the expected format for the evaluator.
 
         Enables some syntactic sugar, such as the use of Python scalars,
         in a manner consistent with the translator. See autocast.py for details.
         """
         return autocast.dynamic_cast_inputs(schema, *inputs)
 
-    def adapt_attributes(self, schema, attributes):
+    def adapt_attributes(
+        self, schema: onnx.defs.OpSchema, attributes: Mapping[str, Any]
+    ) -> tuple[dict[str, Any], dict[str, Any]]:
         """Transform attributes (in-place) to the expected format for the evaluator.
 
         Returns a closure that can be used to evaluate graph-valued attributes.
         """
         use_graph_attribute = self.use_graph_attribute(schema)
         closure = {}
+        adapted_attributes = {}
```

Comment:
May be worth checking if the caller uses the returned value or passed-in parameter. If it is correct, we should update the above comment in line 135 which says attributes are updated in place as well.

Reply:
Updated comment to remove `in-place`